ubiconfig/config_types/flags.py

```diff
@@ -0,0 +1,49 @@
+class Flag:
+    def __init__(self, name, value):
+        self.name = name
+        self.value = _str_to_bool(value)
+
+
+class Flags(object):
+    """group of modules"""
+
+    def __init__(self, flags):
+        """
+        Args:
+            include(list): a list of :class:`Flag` instances
+        """
+        self.flags = flags
+
+    def __getattr__(self, name):
+        # allow getting flags as object attrs
+        for item in self.flags:
+            if item.name == name:
+                return item
+
+    @classmethod
+    def load_from_dict(cls, data):
+        """Create instances of :class:`Flags` from a dictionary
+
+        Args:
+            data(dict): dictionary with data of following format
+
+        .. code-block:: json
+
+            {
+                "flag_1": "value",
+                "flag_2": "value"
+            }
+        """
+        return cls([Flag(name, value) for name, value in data.items()])
+
+
+def _str_to_bool(value):
```

Comment:
If the input string is different that the 4 strings explicitly mentioned in the conditions, it will be returned on the output. In order to always return a boolean like the function name indicates, I'd suggest doing something like this instead:

```
if isinstance(value, str):
    return value.lower() in ("true", "t", "1", "yes", "y")
else:
    raise TypeError('Expected string, got %s' % value)
```

Reply:
Another idea would be to allow any value in the flags and restrict the possible values only via schema, which you also already do. However in this case only the actual boolean true/false would be allowed. But I guess you've already thought of that and found it insufficient in some cases. Is that right? And if so, why?

Reply:
A)
The main reason for doing this is behavior of yaml loaders.
When using BaseLoader boolean-like values are loaded as strings always but on the other hand it correctly parses string that looks as number (e.g.  1.10 needs to parsed as "1.10" but not as 1.1)
When using SafeLoader boolean values are loaded really as booleans but we hit the problem with those number-strings - 1.10 is coerced to 1.1 which would break things.
I'd rather stick with the BaseLoader a do extra conversion for `flags`.
See also this https://github.com/release-engineering/ubi-config/commit/e8679131b6cc4234ae8379755cfbda4bc37940ab

B) 
The intention is also to allow any string value for any future additions but in a case when we can safely say it's a boolean - do conversion.